library/classes.lisp

```diff
@@ -255,11 +255,9 @@
   (define-instance (Into :a :a)
     (define (into x) x))
 
-  (define-class (TryInto :a :b)
-    "TRY-INTO implies some elements of `:a` can be represented exactly by an element of :b, but sometimes not. If not, an error string is returned."
-    ;; Ideally we'd have an associated-type here instead of locking in
-    ;; on String.
-    (tryInto (:a -> Result String :b)))
+  (define-class (TryInto :a :b :c)
```

Comment:
`TryInto :a :b :c (:a :b -> :c)` see [functional dependencies](https://wiki.haskell.org/Functional_dependencies).

Reply:
I just pushed what I believe is more compliant.